src/phoenix/server/api/mutations/user_mutations.py

```diff
@@ -201,6 +207,56 @@ async def patch_viewer(
             await info.context.log_out(user.id)
         return UserMutationPayload(user=to_gql_user(user))
 
+    @strawberry.mutation(
+        permission_classes=[
+            IsNotReadOnly,
+            IsAuthenticated,
+            IsAdmin,
+        ]
+    )  # type: ignore
+    async def delete_users(
+        self,
+        info: Info[Context, None],
+        input: DeleteUsersInput,
+    ) -> None:
+        user_ids = tuple(
+            map(
+                lambda gid: from_global_id_with_expected_type(gid, User.__name__),
+                set(input.user_ids),
+            )
+        )
+        system_role_id = (
+            select(models.UserRole.id)
+            .where(models.UserRole.name == enums.UserRole.SYSTEM.value)
+            .scalar_subquery()
+        )
+        admin_role_id = (
+            select(models.UserRole.id)
```

Comment:
The overall logic may be simplified if we just go ahead and make multiple queries, including one on the `min` admin. The extra cost is negligible because the deletion requests are sporadic events.
```suggestion
            select(min(models.UserRole.id))
```